src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -9,21 +9,19 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
-        public override Task<bool> BindModelAsync(ModelBindingContext bindingContext)
+        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
             if (bindingContext.ModelMetadata.IsReadOnly)
             {
-                return Task.FromResult(false);
+                return Task.FromResult<ModelBindingResult>(null);
```

Comment:
The feedback is primarily based on a recent change to `CompilationResult` that I recently made (@yishaigalatzer's suggestion) that used to return null but now returns a failed status value.